pages/[resourceType]/index.tsx

```diff
@@ -1,85 +1,147 @@
-      fetch(`${process.env.NEXT_PUBLIC_DEQM_SERVER}/${resourceType}`)
-        .then((data) => {
-          return data.json() as Promise<fhirJson.Bundle>;
-        })
-        .then((resourcePageBody) => {
-          setPageBody(resourcePageBody);
-          setFetchingError(false);
-          setLoadingRequest(false);
-        })
-        .catch((error) => {
-          console.log(error.message);
-          setFetchingError(true);
-          setLoadingRequest(false);
-        });
+    let pageUrl = `${resourceType}`;
+    if (page != null) {
+      setActivePageNum(parseInt(page as string));
+      pageUrl = `${resourceType}?page=${page}`;
     }
-  }, [resourceType]);
+    setLoadingRequest(true);
+    fetch(`${process.env.NEXT_PUBLIC_DEQM_SERVER}/${pageUrl}`)
+      .then((data) => {
+        return data.json() as Promise<fhirJson.Bundle>;
+      })
+      .then((resourcePageBody) => {
+        setPageBody(resourcePageBody);
+        setFetchingError(false);
+        setLoadingRequest(false);
+      })
+      .catch((error) => {
+        console.log(error.message);
+        setFetchingError(true);
+        setLoadingRequest(false);
+      });
+  }, [page, resourceType]);
 
-  return loadingRequest ? ( //if loading, Loader object is returned
-    <Center>
-      <div>Loading content...</div>
-      <Loader color="cyan"></Loader>
-    </Center>
-  ) : !fetchingError && pageBody ? ( //if http request was successful, ResourceID component is returned
-    <div>
-      <div
-        style={{
-          paddingRight: "20px",
-        }}
-      >
-        <Stack
+  const updatePageRoute = (pageNum: number) => {
+    router.push(
+      {
+        pathname: `/[resourceType]`,
+        query: {
+          resourceType,
+          page: pageNum,
+        },
+      },
+      `/${resourceType}?page=${pageNum}`,
+      {
+        shallow: true,
+      },
+    );
+  };
+
+  return (
+    <div
+      style={{
+        paddingRight: "20px",
+      }}
+    >
+      <Link href={`${resourceType}/create`} key={`create-${resourceType}`} passHref>
+        <Button
+          component="a"
+          color="cyan"
+          radius="md"
+          size="md"
+          variant="filled"
           style={{
             float: "right",
           }}
         >
-          <Link href={`${resourceType}/create`} key={`create-${resourceType}`} passHref>
-            <Button
-              component="a"
-              color="cyan"
-              radius="md"
-              size="md"
-              variant="filled"
-              style={{
-                float: "right",
-              }}
-            >
-              <div> Create New {`${resourceType}`} </div>
-            </Button>
-          </Link>
-        </Stack>
-      </div>
-      <ResourceIDs jsonBody={pageBody}></ResourceIDs>
+          <div> Create New {`${resourceType}`} </div>
+        </Button>
+      </Link>
+      <h2 style={{ marginTop: "2px", marginBottom: "5px" }}>&nbsp;</h2>
```

Comment:
I really understand why you did this because padding can be such a pain hahaha. Maybe try a group/grid/container something else instead? I feel like there could be something better than a header with a non-breaking space lol but this does work